e2e/common/traits/route_test.go

```diff
@@ -0,0 +1,444 @@
+		certPool := x509.NewCertPool()
+		certPool.AppendCertsFromPEM(certPem)
+		transCfg := &http.Transport{
+			TLSClientConfig: &tls.Config {
+				RootCAs: certPool,
+			},
+		}
+		client = http.Client{Transport: transCfg}	
+	} else {
+		client = http.Client{}
+	}
+	response, err := client.Get(url)
+	defer func() {
+		if response != nil {
+			_ = response.Body.Close()
+		}
+	}()
+	if err != nil {
+		return "", err
+	}
+	assert.Nil(t, err)
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(response.Body)
+	if err != nil {
+		return "", err
+	}
+	assert.Nil(t, err)
+	return buf.String(), nil
+}
+
+func createSecret(ns string) (corev1.Secret, error) {
+	keyCertPair := generateSampleKeyAndCertificate(ns)
+	sec := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: ns,
+			Name:      secretName,
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			corev1.TLSPrivateKeyKey: keyCertPair.Key,
+			corev1.TLSCertKey: keyCertPair.Certificate,
+		},
+	}
+	return sec, TestClient().Create(TestContext, &sec)
+}
+
+func generateSampleKeyAndCertificate(ns string) keyCertificatePair {
+	// Generate the TLS certificate
+	serialNumber := big.NewInt(rand2.Int63())
+	dnsHostname := integrationName + "-" + ns + "." + domainName()
+	x509Certificate := x509.Certificate{
+		SerialNumber: serialNumber,
+		Subject: pkix.Name{
+			Organization: []string{"Camel K test"},
+		},
+		IsCA: 				   true,
+		DNSNames:              []string{dnsHostname},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(1, 0, 0),
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	// generate the private key
+	certPrivateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		fmt.Printf("Error generating private key: %s\n", err)
+	}
+
+	privateKeyBytes := x509.MarshalPKCS1PrivateKey(certPrivateKey)
+	// encode for storing into secret
+	privateKeyPem := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: privateKeyBytes,
+		},
+	)
+	certBytes, err := x509.CreateCertificate(rand.Reader, &x509Certificate, &x509Certificate, &certPrivateKey.PublicKey, certPrivateKey)
+	if err != nil {
+		fmt.Printf("Error generating certificate: %s\n", err)
+	}
+
+	// encode for storing into secret
+	certPem = pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: certBytes,
+	})
+
+	return keyCertificatePair {
+		Key: privateKeyPem,
+		Certificate: certPem,
+	}
+}
+
+func domainName() string {
+	consoleroute := Route("openshift-console", "console")()
+	return strings.ReplaceAll(consoleroute.Spec.Host, "console-openshift-console.", "")
+}
+
+func getService(ns string, name string) (*corev1.Service, error) {
```

Comment:
If it's a generic method, and is not already present in test_support.go, we better move it there and have it available for anybody willing to use it.

Reply:
Dropped this `getService` in favor of using `Service` from test_support.go.